modules/SofaGuiCommon/src/sofa/gui/BatchGUI.cpp

```diff
@@ -74,8 +74,18 @@ int BatchGUI::mainLoop()
             if (i != nbIter)
             {
                 sofa::helper::AdvancedTimer::begin("Animate");
+
                 sofa::simulation::getSimulation()->animate(groot.get());
-                sofa::helper::AdvancedTimer::end("Animate");
+
+                const auto timerOutputStr = sofa::helper::AdvancedTimer::end("Animate", groot->getTime(), groot->getDt());
+                if (!timerOutputStr.empty() && timerOutputStr.compare("null") != 0)
+                {
+                    std::string jsonFilename = sofa::helper::system::SetDirectory::GetFileNameWithoutExtension(filename.c_str()) + "_" + std::to_string(i) + ".json";
+                    msg_info("BatchGUI") << "Writing " << jsonFilename;
+                    std::ofstream out(jsonFilename);
+                    out << timerOutputStr;
+                    out.close();
+                }
```

Comment:
This makes me think that a json file will be created everytime we execute runSofa in batch mode, wether or not we use `--computationTimeOutputType=json`.
What do I misunderstand?

Reply:
If you don't set `--computationTimeSampling`, `timerOutputStr` is always empty.
If you set `--computationTimeOutputType=stdout` and `--computationTimeSampling`, `timerOutputStr` is always empty.
A json file is saved iff `--computationTimeOutputType=json` (or ljson) and `--computationTimeSampling` is set.